src/com/owncloud/android/ui/fragment/PublicShareDialogFragment.java

```diff
@@ -239,7 +240,10 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
             mNameValueEdit.setText(mPublicShare.getName());
 
             switch (mPublicShare.getPermissions()) {
-                case (UPDATE_PERMISSION | CREATE_PERMISSION):
+                case (OCShare.CREATE_PERMISSION_FLAG
+                        | OCShare.DELETE_PERMISSION_FLAG
+                        | OCShare.UPDATE_PERMISSION_FLAG
+                        | OCShare.READ_PERMISSION_FLAG):
                     mReadWriteButton.setChecked(true);
                     break;
                 case CREATE_PERMISSION:
```

Comment:
Please use `OCShare.CREATE_PERMISSION_FLAG` as you did in the previous case. Besides, delete the constants `CREATE_PERMISSION` and `UPDATE_PERMISSION` defined at the top of the class since you are using directly `OCShare.whatever`

Reply:
fixed